packages/fluentui/react-northstar/src/components/Tree/hooks/useSelectableTree.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import { TreeProps } from '../Tree';
+import { BaseFlatTreeItem } from './flattenTree';
+import { useTreeSelectState } from './useTreeSelectState';
+import { useTree } from './useTree';
+
+export type SelectableFlatTreeItem = BaseFlatTreeItem & {
+  /**
+   * when selected=true, the tree item is fully selected, indicating all its descendents are fully selected;
+   * when selected=false, the tree item is not selected, indicating none of its descendents is selected;
+   * when selected='indeterminate', only part of the tree item's descendents are selected
+   */
+  selected?: boolean | 'indeterminate';
+};
+
+export function useSelectableTree(props: TreeProps) {
+  const baseTree = useTree(props);
+  const { flatTree: baseflatTree, tobeRenderedItemsProps } = baseTree;
+
+  const { selectedItemIds, flatTree, toggleSelect } = useTreeSelectState(props, baseflatTree);
+
+  React.useMemo(() => {
+    tobeRenderedItemsProps.forEach(itemProps => {
+      // tree item is selectable when it is in a selectable tree, and does not receive selectable=false prop
+      itemProps.selectable = props.selectable ? itemProps.selectable !== false : false;
```

Comment:
By transitivity

(false !== false) === false
(true !== false) === true

```suggestion
      itemProps.selectable = props.selectable ? itemProps.selectable : false;
```

Reply:
Should also work

```suggestion
      itemProps.selectable = props.selectable && itemProps.selectable
```

Reply:
I don't think this produce the same result.

When `props.selectable=true` (this specifies if tree is selectable or not), by default `itemProps.selectable` (this specifies if tree item is selectable or not) should be true, unless the input explicitly specifies `itemProps.selectable=false`, so it goes:

```
props.selectable | itemProps.selectable | result
------------------------------------------------
true             | true                 | true
                 | undefined            | true
                 | false                | false 
```

Reply:
oh I see so `itemProps.selectable` can be undefined, then it does make sense. My bad